### DISCUSSION TITLE
add Watermark  in html

### DISCUSSION THREAD

**Discussion by xiaozhiagi** [zauberzeug/nicegui#1332] [Q&A] [answered]
### Question

how to add text Watermark  in html.

**Comment by rodja** [comment]
Do you mean some overlay text like this?

```py
ui.label('watermark').classes('absolute top-1/2 left-1/2 transform -translate-x-1/2 -translate-y-1/2 '
                              'rotate-45 opacity-50 text-5xl text-black z-0 pointer-events-none')
```

**Reply by xiaozhiagi** [reply]
Sorry. Let me describe my requirement: I want multiple tilted watermarks (top, middle, bottom) evenly distributed throughout the entire HTML, and the watermarks are fixed without the need to scroll with the content. The following code is a demo, but I won't achieve the effect of multiple watermarks:

```py
ui.add_body_html('''
    <style>
    body {
        position: relative;
    }
    body::after {
        content: "Watermark Text";
        position: fixed;
        top: 50%;
        left: 50%;
        transform: translate(-50%, -50%);
        font-size: 32px;
        color: rgba(0, 0, 0, 0.1);  /* 设置水印文字颜色与透明度 */
        z-index: -1;
    }
</style>
    ''')
```

**Comment by falkoschindler** [comment, marked as answer]
Here is an example with a grid of 3x3 watermarks. You can adjust it to your needs:
```py
for x in [25, 50, 75]:
    for y in [25, 50, 75]:
        ui.label('NiceGUI').classes(f'fixed left-[{x}%] top-[{y}%] -translate-x-1/2 -translate-y-1/2 '
                                    '-rotate-45 opacity-10 text-5xl z-0 pointer-events-none')

for _ in range(100):
    ui.label('Hello World! ' * 100)  # some dummy content
```

![127 0 0 1_1234_](https://github.com/zauberzeug/nicegui/assets/5767091/bf4a1c4b-75da-4d31-8ccb-c1c054532338)

**Reply by xiaozhiagi** [reply]
Okay, currently we have found an issue with the watermark in ui.left_ The drawer and ui.card are not working and have been overwritten. How can I make the watermark appear in these ui.lefts_ Watermarks can still be displayed in the background of drawer and ui.card

**Reply by falkoschindler** [reply]
I'm sorry, but it's difficult for me to fully understand the issue without seeing any code or screenshots. These could provide valuable context that could aid in diagnosing the problem.

In general, providing all the relevant information when you initially post the question would greatly streamline the process. This way, we can avoid a situation where we're gradually formulating responses and you're continually adding new requirements. This iterative approach can be quite time-consuming and may exceed our capacity.

We're eager to help you solve this problem, so please consider providing as much detail as possible upfront. Thank you!

**Reply by xiaozhiagi** [reply]
How to implement "logo image+text watermark"？

**Reply by rodja** [reply]
Instead of `ui.label('NiceGUI')` form @falkoschindler's example you can use `ui.row` or similar to group multiple elements.

**Reply by xiaozhiagi** [reply]
Can you give an example using code

**Reply by rodja** [reply]
```py
for x in [25, 50, 75]:
    for y in [25, 50, 75]:
        with ui.column().classes(f'fixed left-[{x}%] top-[{y}%] -translate-x-1/2 -translate-y-1/2 '
                                 '-rotate-45 opacity-10 text-5xl z-0 pointer-events-none'):
            ui.label('NiceGUI')
            ui.label('rocks!')
```
